app/packages/looker-3d/src/fo3d/Gizmos.tsx

```diff
@@ -99,34 +106,35 @@ export const Gizmos = () => {
     return Math.max(sceneSize.x, sceneSize.y);
   }, [sceneBoundingBox, upVector]);
 
-  // cell size is based on max bounding box size
-  // following multipliers (0.5, 10, 30) are arbitrary and can be adjusted
-  const cellSize = useMemo(
-    () => maxInOrthonormalPlane * 0.5,
-    [maxInOrthonormalPlane]
-  );
-  const sectionSize = useMemo(() => cellSize * 10, [cellSize]);
+  const cellSize = useRecoilValue(gridCellSizeAtom);
+  const sectionSize = useRecoilValue(gridSectionSizeAtom);
+  const isGridInfinitelyLarge = useRecoilValue(isGridInfinitelyLargeAtom);
+  const shouldFade = useRecoilValue(shouldGridFadeAtom);
+  const gridSize = useRecoilValue(gridSizeAtom);
 
-  const fadeDistance = useMemo(
-    () => maxInOrthonormalPlane * 30,
-    [maxInOrthonormalPlane]
-  );
+  // The fade distance is the distance at which the grid will start to fade out
+  // the multipliers and offset are arbitrary
+  const fadeDistance = useMemo(() => {
+    return maxInOrthonormalPlane * 10 + 100;
+  }, [maxInOrthonormalPlane, gridSize, isGridInfinitelyLarge]);
 
   return (
     <>
       {isGridOn && (
         <>
           <Grid
             quaternion={gridHelperQuarternion}
-            infiniteGrid
+            infiniteGrid={isGridInfinitelyLarge}
             side={DoubleSide}
+            args={[gridSize, gridSize]}
             cellSize={cellSize}
             sectionSize={sectionSize}
             sectionColor={GRID_SECTION_COLOR}
             cellColor={GRID_CELL_COLOR}
             fadeDistance={fadeDistance}
-            fadeStrength={1}
-            followCamera
+            fadeStrength={shouldFade ? 1 : 0}
+            followCamera={shouldFade}
+            fadeFrom={0.5}
```

Comment:
This is a lot more configurable than blender... lol

<img width="297" alt="image" src="https://github.com/voxel51/fiftyone/assets/462228/2d3664a7-54e0-4716-9257-ad0865df178e">

Reply:
Yeah... 😅 
We expose "fading", "infinite / non-infinite", "cell size", "section size", and "grid size" if non-infinite.